From 9241314d6a7a2197b20918ad8cc9740797658048 Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Wed, 21 May 2014 17:26:54 +0800
Subject: [PATCH 259/280] ENGR00314307 Remove setrobust on Android

Hot fix to ENGR00314307.
Remove pthread_mutexattr_setrobust on Android due to build break.

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 vpu/vpu_util.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/vpu/vpu_util.c b/vpu/vpu_util.c
index 12bbaff..bc5bea7 100644
--- a/vpu/vpu_util.c
+++ b/vpu/vpu_util.c
@@ -1195,7 +1195,9 @@ semaphore_t *vpu_semaphore_open(void)
 		dprintf(4, "sema not init\n");
 		pthread_mutexattr_init(&psharedm);
 		pthread_mutexattr_setpshared(&psharedm, PTHREAD_PROCESS_SHARED);
+#ifndef BUILD_FOR_ANDROID
 		pthread_mutexattr_setrobust(&psharedm, PTHREAD_MUTEX_ROBUST);
+#endif
 #ifdef FIFO_MUTEX
 		pthread_mutex_init(&semap->api_lock.mutex, &psharedm);
 		pthread_condattr_init(&psharedc);
@@ -1393,10 +1395,12 @@ unsigned char semaphore_wait(semaphore_t *semap, int mutex)
 #else
 	{
 		ret = pthread_mutex_timedlock(&semap->api_lock, &ts);
+#ifndef BUILD_FOR_ANDROID
 		if (ret == EOWNERDEAD) {
 			pthread_mutex_consistent(&semap->api_lock);
 			ret = 0;
 		}
+#endif
 	}
 #endif
 	else if (mutex == REG_MUTEX)
-- 
1.8.0

